File: sort/bubble_sort_optimized/php/bubble_sort_optimized.php
<?php

public function bubbleSortOptimized($arr, $n) {
  for($i = 0; $i < $n; $i++) {  
    $flag = false;
    for($j = 0; $j < $n - $i - 1; $j++)
     {
        if($arr[$j] > $arr[$j + 1]) {
          $flag        = true;
          $temp        = $arr[$j + 1];
          $arr[$j + 1] = $arr[$j];
          $arr[$j]     = $temp;
        }
     }
    // No Swapping happened, array is sorted
    if(!$flag){ 
       return; 
    } 
 }
}

File: sort/bubble_sort_optimized/go/bubble_sort_optimized.go
package bubblesort

// BubbleSort Optimized implementation of bubble sort
func BubbleSort(arr []int) {
	for i := range arr[:len(arr)] {
		flag := false
		for j := range arr[:len(arr)-i-1] {
			if arr[j] > arr[j+1] {
				flag = true
				temp := arr[j+1]
				arr[j+1] = arr[j]
				arr[j] = temp
			}
		}
		// No Swapping happened, array is sorted
		if !flag {
			return
		}
	}
}
